Remove UA styles for h1 in article, aside, nav, and section

annevk opened the pull request:
#### 2e8474128f34d8d8214ef12d0f9fb632733da98d
<pre>
Remove UA styles for h1 in article, aside, nav, and section
<a href="https://bugs.webkit.org/show_bug.cgi?id=292765">https://bugs.webkit.org/show_bug.cgi?id=292765</a>
<a href="https://rdar.apple.com/151455566">rdar://151455566</a>

Reviewed by Darin Adler.

Firefox has shipped this.

Chrome has also taken steps in deprecating this pattern and plans to
follow Firefox.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/ca3b40b0dcd17406d6997dcb6285227af30a8378">https://github.com/web-platform-tests/wpt/commit/ca3b40b0dcd17406d6997dcb6285227af30a8378</a>

Canonical link: <a href="https://commits.webkit.org/297671@main">https://commits.webkit.org/297671@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/98d0c1ed4809a34d346f336c873caa00f237068b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112518 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32250 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22728 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118716 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62971 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/114480 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32902 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40813 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85633 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/36247 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115465 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26254 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101223 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65935 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25552 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19356 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62476 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95644 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19430 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121938 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39592 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29483 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94498 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39974 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97456 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94238 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24055 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39351 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17154 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/35669 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39480 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44968 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39117 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42452 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40858 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
I was reluctant to approve because I couldn't see any browser shipping this, nor did I see the spec/WPT PR being merged.

but after some research, I found Firefox is shipping it:
https://bugzilla.mozilla.org/show_bug.cgi?id=1964922 

though I would like to see the spec PR being moved forward ideally.

annevk committed:
Remove UA styles for h1 in article, aside, nav, and section
https://bugs.webkit.org/show_bug.cgi?id=292765
rdar://151455566

Reviewed by Darin Adler.

Firefox has shipped this.

Chrome has also taken steps in deprecating this pattern and plans to
follow Firefox.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/ca3b40b0dcd17406d6997dcb6285227af30a8378

Canonical link: https://commits.webkit.org/297671@main
